c_src/erl_optics.c

```
@@ -46,14 +48,49 @@ static int load(ErlNifEnv *env, void **priv_data, ERL_NIF_TERM load_info) {
     atom_dist = enif_make_atom(env, "dist");
     atom_histo = enif_make_atom(env, "histo");
     atom_quantile = enif_make_atom(env, "quantile");
-    poller = optics_poller_alloc();
-    ctx = enif_alloc(sizeof(struct map_ctx));
-    if (poller) {
-      optics_poller_backend(poller, (void *) ctx, backend_eo, backend_free_eo);
-    }
+    carbon_poller = 0;
+    erlang_poller = 0;
     return 0;
 }
 
+
+static ERL_NIF_TERM eo_allocate_carbon_poller(
+    ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
+{
+    if (!poller){
+        poller = optics_poller_alloc();
+    }
+    if (!poller) return ERROR("poller_allocation_error");
+    if (carbon_poller == 0) {
+        size_t host_size = 255;
+        size_t port_size = 255;
+        char host[host_size];
+        char port[port_size];
+        enif_get_string(env, argv[1], host, host_size, ERL_NIF_LATIN1);
```

I think this could use some error checking, and maybe return an error